behaviour/graql/language/insert.feature

```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
+    When get answers of graql query
+      """
+      insert
+      $x isa person, has name "Bruce Wayne", has ref 0;
+      $z isa company, has name "Wayne Enterprises", has ref 0;
+      """
+    When the integrity is validated
+    Then answer size is: 1
+    Then concept identifiers are
+      |     | check | value |
+      | BRU | key   | ref:0 |
+      | WAY | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   | z   |
+      | BRU | WAY |
+
+
+  Scenario: when inserting a thing variable with a type variable, the answer contains both variables
+    When get answers of graql query
+      """
+      match
+        $type type company;
+      insert
+        $x isa $type, has name "Microsoft", has ref 0;
+      """
+    When the integrity is validated
+    Then concept identifiers are
+      |     | check | value   |
+      | MIC | key   | ref:0   |
+      | COM | label | company |
+    Then uniquely identify answer concepts
+      | x   | type |
+      | MIC | COM  |
+
+
+  ################
+  # MATCH-INSERT #
+  ################
+
+  Scenario: match-insert triggers one insert per answer of the match clause
+    Given graql define
+      """
+      define
+      language sub entity, has name, has is-cool, key ref;
+      is-cool sub attribute, value boolean;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa language, has name "Norwegian", has ref 0;
+      $y isa language, has name "Danish", has ref 1;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      match
+        $x isa language;
+      insert
+        $x has is-cool true;
+      """
+    When the integrity is validated
+    Then get answers of graql query
+      """
+      match $x has is-cool true; get;
+      """
+    Then concept identifiers are
+      |     | check | value |
+      | NOR | key   | ref:0 |
+      | DAN | key   | ref:1 |
+
+
+  Scenario: the answers of a match-insert only include the variables referenced in the `insert` block
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Eric", has ref 0;
+      $y isa company, has name "Microsoft", has ref 1;
+      $r (employee: $x, employer: $y) isa employment, has ref 2;
+      $z isa person, has name "Tarja", has ref 3;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match
+        (employer: $x, employee: $z) isa employment;
+        $y isa person, has name "Tarja";
+      insert
+        (employer: $x, employee: $y) isa employment;
+      """
+    Then concept identifiers are
+      |     | check | value |
+      | MIC | key   | ref:1 |
+      | TAR | key   | ref:3 |
+    # Should only contain variables mentioned in the insert (so excludes `$z`)
+    Then uniquely identify answer concepts
+      | x   | y   |
+      | MIC | TAR |
+
+
+  Scenario: if match-insert matches nothing, then it inserts nothing
```

Comment:
`then nothing is inserted` - lets leave out "it"

Reply:
done